### ISSUE TITLE
Fixing asymmetric IDataset.getMetadata

### ISSUE THREAD

**Issue by IanMayo** [eclipse/january#30]
IDataset has getMetadata() and setMetadata() methods.

But there are asymmetric:

````
  void setMetadata(MetadataType metadata);
  IMetadata getMetadata();
````

My colleague has mistaken how to use metadata, in almost exactly the same way I did back in Sep 2015.

@PeterC-DLS  previously indicated this could be deleted:

````
>
> * I accidentally conflated the MetadataType and IMetadata types
> handled by ILazyDataset & IDataset whilst familiarising myself with
> the code.    Since IMetadata actually is an instance of MetadataType,
> I guess my confusion would be avoided by removing getMetadata()
> from IDataset (replacing it with getMetadata(IMetadata.class).

It was legacy that may be worth removing...
````

I'm happy to start work on a PR.

Note: I don't believe new metadata types should extend [IMetadata](https://github.com/eclipse/january/blob/master/org.eclipse.january/src/org/eclipse/january/metadata/IMetadata.java), since it requires methods unrelated to general metadata.

**Comment by IanMayo** [comment]
Fixed provided in PR #31

**Comment by PeterC-DLS** [comment]
I think there's a misunderstanding here: `IMetadata` is a sub-interface of `MetadataType` and not all metadata interfaces are sub-interfaces of it.

I meant we should deprecated `getMetadata()` and replace those users with `getFirstMetadata(IMetadata.class)`.

**Comment by PeterC-DLS** [comment]
Note, `IMetadata` is historically bad name :blush:. It could rename as `DataFileMetadata`...

**Comment by IanMayo** [comment]
Ok, deprecating ````getMetadata()```` would avoid the confusion we experienced,

Given that the implementation now supports multiple metadata entries per dataset, would ````addMetadata(obj)```` be naming than ````setMetadata(obj)````?

If the DAWN usage of ````setMetadata()```` is too extensive to allow renaming, I guess ````setMetadata()```` could be marked as deprecated, and under-the-covers actually call ````addMetadata()````.

Happy to implement this if you wish.

**Comment by PeterC-DLS** [comment]
The `addMetadata` and `setMetadata` methods are distinct in that we support lists of metadata of various types and the former adds to a list whereas the latter sets the list to a single item.

We may want to reconsider whether having more than one metadata (of a given type) is necessary. I not sure it is and if not, then the distinction noted above becomes redundant.

**Comment by IanMayo** [comment]
Aah, ok. Will submit PR to deprecate ````getMetadata()````, and adopt your recommended strategy.

**Comment by IanMayo** [comment]
Issue resolved, in #33
